### PR TITLE
support lists of lists of lists

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/list_of_lists/test/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "scripts": {
+        "test": "cd .. && ts-node --transpile-only --ignore=false test/test.ts"
+    },
+    "dependencies": {
+        "azle": "github:demergent-labs/azle#kybra_test_reuse",
+        "fast-equals": "4.0.3"
```
Comment (lastmjs) [inline examples/list_of_lists/test/package.json:7]: If you could make this a dev dependency, `npm install fast-equals --save-dev`.

File: examples/list_of_lists/test/test.ts [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+    | Int8Array
+    | Int16Array
+    | Int32Array
+    | BigInt64Array
+    | Uint8Array
+    | Uint16Array
+    | Uint32Array
+    | BigUint64Array;
+
+const tests: Test[] = [
```
Comment (lastmjs) [inline examples/list_of_lists/test/test.ts:42]: Can you put these tests in a sibling file called `tests.ts` and import them?

File: examples/list_of_lists/test/test.ts [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+import { deepEqual } from 'fast-equals';
+import { Opt } from 'azle';
+
+const list_of_lists_canister = createActor('rrkah-fqaaa-aaaaa-aaaaq-cai', {
+    agentOptions: {
+        host: 'http://127.0.0.1:8000'
+    }
+});
+
+function arrEqual<T>(a: DeepArray<T>, b: DeepArray<T>): boolean {
```
Comment (lastmjs) [inline examples/list_of_lists/test/test.ts:15]: Can you put this stuff below the tests? And this should move over into the `tests.ts` file.

File: kybra/compiler/kybra_generate/src/generators/vm_value_conversion/try_from_vm_value/vec.rs [hunk 4]
```diff
@@ -126,28 +84,43 @@ pub fn generate_vec_impls() -> proc_macro2::TokenStream {
             }
         }
 
-        // TODO consider creating a macro that can derive Vec of Vec multiple levels deep for any type
-        impl CdkActTryFromVmValue<Vec<Vec<u8>>, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef {
-            fn try_from_vm_value(self, vm: &rustpython::vm::VirtualMachine) -> Result<Vec<Vec<u8>>, CdkActTryFromVmValueError> {
-                try_from_vm_value_generic_array(self, vm)
-            }
+        trait KybraTryFrom {
```
Comment (lastmjs) [inline kybra/compiler/kybra_generate/src/generators/vm_value_conversion/try_from_vm_value/vec.rs:87]: We should discuss the naming of this trait

File: kybra/compiler/kybra_generate/src/generators/vm_value_conversion/try_from_vm_value/vec.rs [hunk 5]
```diff
@@ -126,28 +84,43 @@ pub fn generate_vec_impls() -> proc_macro2::TokenStream {
...
         }
 
-        impl<T> CdkActTryFromVmValue<Vec<Box<T>>, &rustpython::vm::VirtualMachine> for rustpython::vm::PyObjectRef
-        where
-            rustpython::vm::PyObjectRef: for<'a> CdkActTryFromVmValue<T, &'a rustpython::vm::VirtualMachine>
-        {
-            fn try_from_vm_value(self, vm: &rustpython::vm::VirtualMachine) -> Result<Vec<Box<T>>, CdkActTryFromVmValueError> {
-                try_from_vm_value_generic_array::<Box<T>>(self, vm)
-            }
+        impl<T> KybraTryFrom for Vec<T> {
```
Comment (lastmjs) [inline kybra/compiler/kybra_generate/src/generators/vm_value_conversion/try_from_vm_value/vec.rs:90]: I need to review these two files more deeply

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/list_of_lists/test/package.json (hunk 1 above)
Comment (bdemann) [reply examples/list_of_lists/test/package.json:7]: I did this in azle, so I don't think we will need it here once azle is merged in

File: examples/list_of_lists/test/test.ts (hunk 2 above)
Comment (bdemann) [reply examples/list_of_lists/test/test.ts:42]: This is how it is in azle and I will just get rid of the tests once we have it merged in azle

File: examples/list_of_lists/test/test.ts (hunk 3 above)
Comment (bdemann) [reply examples/list_of_lists/test/test.ts:15]: I have it that way in azle

**Pull request merged**